File: frontend/src/app/features/public/model/PhotoDTO.ts
export interface PhotoDTO {
    Id: number;
    Title: string;
    Description: string;
    LikeCount: number;
    ViewCount: number;
    Price: number;
    Location: string;
    User?: User;
    Approved: boolean;
    CreateAt: Date;
    Orientation?: string;
    Url: string;
    Size?: string;
    IsLiked: boolean;
  }
  

export interface User {
    Id: number;
    Username: string;
    FirstName: string;
    LastName: string;
    Email: string;
    ProfilePicture?: string;
  }
  

File: backend/backend/Helper/DTO-s/PhotoDTO.cs
using backend.Data.Models;

namespace backend.Helper.DTO_s
{
    public class PhotoDTO
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int LikeCount { get; set; }
        public int ViewCount { get; set; }
        public int Price { get; set; }
        public string Location { get; set; }
        public UserDTO? User { get; set; }
        public bool Approved { get; set; }
        public DateTime CreateAt { get; set; }
        public string? Orientation { get; set; }
        public string Url { get; set; }
        public string? Size { get; set; }

        public bool IsLiked { get; set; }

        public List<string> Tags { get; set; } = new List<string>();
    }
}
